### PR TITLE
LDAP group membership

### PR DESCRIPTION
**Pull request by amendlik**
### What does this PR do?
Fix a regression in the LDAP auth provider

### What issues does this PR fix or reference?
The changes made in #42280 and #42426 impacted the code for determining group membership. This change allows an authenticated LDAP bind when getting group members.

### Tests written?
No

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/auth/ldap.py [hunk 1]
```diff
@@ -316,8 +321,13 @@ def groups(username, **kwargs):
     '''
     group_list = []
 
-    # Perform un-authenticated bind to determine group membership
-    bind = _bind_for_search(anonymous=_config('anonymous', mandatory=False))
+    # If bind credentials are configured, use them instead of user's
+    if _config('binddn', mandatory=False) and _config('bindpw', mandatory=False):
+        bind = _bind_for_search(anonymous=_config('anonymous', mandatory=False))
+    else:
+        bind = _bind(username, kwargs['password'],
```
Comment (DmitryKuzmenko) [inline salt/auth/ldap.py]: I'd better like to either use `kwargs.get` here to avoid potential KeyError raising.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
Looks good at all except the requested minor change. After the change I'm OK as long as all the tests pass.

File: salt/auth/ldap.py (hunk 1 above)
Comment (amendlik) [reply salt/auth/ldap.py]: I've made the requested change.

**Review by DmitryKuzmenko — APPROVED** [review]
LGTM

**Review by rallytime — APPROVED** [review]

**Review by gtmanfred — APPROVED** [review]

**Pull request merged**